src/overlays/actors/ovl_En_Rat/z_en_rat.c

```diff
@@ -28,123 +47,913 @@ const ActorInit En_Rat_InitVars = {
+    if (EN_RAT_IS_OVERWORLD_TYPE(&this->actor)) {
+        this->actor.params = EN_RAT_TYPE_OVERWORLD;
+    } else {
+        this->actor.params = EN_RAT_TYPE_DUNGEON;
+    }
+
+    SkelAnime_InitFlex(play, &this->skelAnime, &gRealBombchuSkel, &gRealBombchuRunAnim, this->jointTable,
+                       this->morphTable, REAL_BOMBCHU_LIMB_MAX);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 25.0f);
+
+    if (EN_RAT_GET_TYPE(&this->actor) == EN_RAT_TYPE_DUNGEON) {
+        Effect_Add(play, &this->blure1Index, EFFECT_BLURE2, 0, 0, &sBlureInit);
+        Effect_Add(play, &this->blure2Index, EFFECT_BLURE2, 0, 0, &sBlureInit);
+        this->timer = 30;
+    } else {
+        this->timer = 150;
+    }
+
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    EnRat_InitializeAxes(this);
+    EnRat_UpdateRotation(this);
+
+    if ((attackRange == 255) || (attackRange == 0)) {
+        this->attackRange = 350.0f;
+    } else if (EN_RAT_GET_TYPE(&this->actor) == EN_RAT_TYPE_DUNGEON) {
+        this->attackRange = attackRange * 0.1f * 40.0f;
+    } else {
+        this->attackRange = attackRange * 0.5f * 40.0f;
+    }
+
+    if (!sTexturesDesegmented) {
+        for (i = 0; i < ARRAY_COUNT(sSparkTextures); i++) {
+            sSparkTextures[i] = Lib_SegmentedToVirtual(sSparkTextures[i]);
+        }
+
+        sTexturesDesegmented = true;
+    }
+
+    EnRat_SetupIdle(this);
+}
+
+void EnRat_Destroy(Actor* thisx, PlayState* play) {
+    EnRat* this = THIS;
+
+    if (EN_RAT_GET_TYPE(&this->actor) == EN_RAT_TYPE_DUNGEON) {
+        Effect_Destroy(play, this->blure1Index);
+        Effect_Destroy(play, this->blure2Index);
+    }
+
+    Collider_DestroySphere(play, &this->collider);
+}
+
+void EnRat_InitializeAxes(EnRat* this) {
+    Matrix_RotateYS(this->actor.shape.rot.y, MTXMODE_NEW);
+    Matrix_RotateXS(this->actor.shape.rot.x, MTXMODE_APPLY);
+    Matrix_RotateZS(this->actor.shape.rot.z, MTXMODE_APPLY);
+
+    Matrix_MultVecZ(1.0f, &this->axisForwards);
+    Matrix_MultVecY(1.0f, &this->axisUp);
+    Matrix_MultVecX(1.0f, &this->axisLeft);
+}
+
+/**
+ * Returns true if floorPoly is valid for the Real Bombchu to move on, false otherwise.
+ */
+s32 EnRat_UpdateFloorPoly(EnRat* this, CollisionPoly* floorPoly, PlayState* play) {
+    Vec3f normal;
+    Vec3f vec;
+    f32 angle;
+    f32 magnitude;
+    f32 normDotUp;
+
+    this->actor.floorPoly = floorPoly;
+
+    if (floorPoly != NULL) {
+        normal.x = COLPOLY_GET_NORMAL(floorPoly->normal.x);
+        normal.y = COLPOLY_GET_NORMAL(floorPoly->normal.y);
+        normal.z = COLPOLY_GET_NORMAL(floorPoly->normal.z);
+    } else {
+        normal.x = 0.0f;
+        normal.z = 0.0f;
+        normal.y = 1.0f;
+    }
+
+    normDotUp = DOTXYZ(normal, this->axisUp);
+
+    if (fabsf(normDotUp) >= 0.999f) {
+        return false;
+    }
+
+    angle = func_80086C48(normDotUp);
+    if (angle < 0.001f) {
+        return false;
+    }
```

Comment:
```suggestion
    normDotUp = DOTXYZ(normal, this->axisUp);
    if (fabsf(normDotUp) >= 0.999f) {
        return false;
    }

    angle = func_80086C48(normDotUp);
    if (angle < 0.001f) {
        return false;
    }
```
Just for consistency